### PR TITLE
Feat/ecs/bullet systems

### PR DESCRIPTION
**Pull request by aurelenc**
Add shoot systems
+ shoot from client and manage bullets in server
+ weapon class
+ keep shooting while the space bar is pressed
#
+ crossplatform paths
+ map size

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,4 +34,8 @@ add_subdirectory(src)
 if (BUILD_TEST_SUITE)
      enable_testing()
      add_subdirectory(tests)
+     endif()
+
+if (CLIENT_COMPILATION_MODE)
+     add_compile_definitions(CLIENT_COMPILATION_MODE)
```
Comment (Cavonstavant) [inline CMakeLists.txt:40]: > For future use

Use `target_compile_definition()`  in order to set `CLIENT_MODE` only for the client target

File: src/ecs/World.hpp [hunk 2]
```diff
@@ -50,7 +46,7 @@ namespace ecs
          */
         void addSystem(const std::function<void(World &)> &system) { _systems.push_back(system); }
 
-        sf::RenderWindow &getWindow() { return *_window; }
+        utils::Window &getWindow() { return *_window; }
```
Comment (Cavonstavant) [inline src/ecs/World.hpp:49]:
```suggestion
        inline utils::Window &getWindow() { return *_window; }
```

**Review by Cavonstavant — CHANGES_REQUESTED** [review]
minor changes needed. Everythings okay otherwise

**Review by Bootoyka — APPROVED** [review]
lgtm

File: CMakeLists.txt (hunk 1 above)
Comment (Bootoyka) [reply CMakeLists.txt:40]: @aurelenc

Comment (aurelenc) [reply CMakeLists.txt:40]: why not add_compile_definitions only on the client's cmakelists ? @Cavonstavant

Comment (aurelenc) [reply CMakeLists.txt:40]: demo will probably be deleted once the game is successfully networked

**Review by Cavonstavant — APPROVED** [review]
Ok

File: CMakeLists.txt (hunk 1 above)
Comment (Cavonstavant) [reply CMakeLists.txt]: @aurelenc Yes ofc. We can use target_compile_definitions() to add a def. to specific target

**Review by Bootoyka — APPROVED** [review]
lgtm

**Comment by Bootoyka** [comment]
Good progress

**Pull request merged**